cmd/proxy/Dockerfile

```diff
@@ -1,15 +1,30 @@
-FROM golang:1.11
+FROM golang:1.11-alpine AS builder
 
-RUN mkdir -p $GOPATH/src/github.com/gomods/athens
 WORKDIR $GOPATH/src/github.com/gomods/athens
 
 COPY . .
 
-RUN CGO_ENABLED=0 GO111MODULE=on go build -mod=vendor -o /bin/app ./cmd/proxy
-RUN ./scripts/create_default_config.sh
+RUN apk update && \ 
+	apk add --no-cache bash && \
+	GO111MODULE=on CGO_ENABLED=0 go build -mod=vendor -o /bin/app ./cmd/proxy && \
+	./scripts/create_default_config.sh
+
+FROM alpine
+
+ENV GO111MODULE=on
+
+COPY --from=builder /bin/app /bin/app
+COPY --from=builder /go/src/github.com/gomods/athens/config.toml /config/config.toml
+COPY --from=builder /go/src/github.com/gomods/athens/cmd/proxy/locales /go/src/github.com/gomods/athens/cmd/proxy/locales
+COPY --from=builder /go/src/github.com/gomods/athens/cmd/proxy/assets /go/src/github.com/gomods/athens/cmd/proxy/assets
+COPY --from=builder /usr/local/go/bin/go /bin/go
+
+RUN apk update &&  \
+		apk add --no-cache bzr git mercurial openssh-client subversion procps && \
+		mkdir -p /usr/local/go
 
 ENV GO_ENV=production
 
 EXPOSE 3000
 
-ENTRYPOINT /bin/app -config_file=config.toml
+ENTRYPOINT ["app", "-config_file=/config/config.toml"]
```

Comment:
shouldn't this be "/bin/app"?

Reply:
bin is on the PATH so it's not needed.

Reply:
I think we should not use the generic name `app`, it is better to use names such as `athens` or `athens-proxy`.
`ENTRYPOINT` is usually used to execute pre-launch scripts. `CMD` is more suitable for this.

Reply:
@tnextday agreed. However, I think I'll approve and merge this for now and submit a change afterwards. @gcstang has gracefully kept up with all of our comments and it looks good enough for this pr. Thank you both!